bug(admin): offline storage logs endpoint 500s on default empty-search load

## Problem
Trang [https://holilihu.online/admin/offline-storage](https://holilihu.online/admin/offline-storage) đang hiển thị như thể production không có bản ghi telemetry, nhưng endpoint logs phía sau trang này thực tế bị `500 INTERNAL_ERROR` khi load với bộ lọc mặc định (`search` rỗng / `null`).

Điều này làm màn quan sát offline storage đánh lừa admin vận hành: analytics card vẫn render bình thường, còn khu vực bảng log lại rơi về empty state giống hệt trường hợp “không có dữ liệu”.

## Production Evidence
Quan sát ngày `2026-04-23` (múi giờ Asia/Saigon) bằng tài khoản `ADMIN` trên production:

1. Mở `https://holilihu.online/admin/offline-storage`
2. Trang render đầy đủ shell + analytics card + empty state `Chưa có bản ghi phù hợp`
3. Gọi trực tiếp API bằng cùng bearer token admin:
   - `GET /api/v3/admin/client-telemetry/offline-storage?page=0&size=5` -> `500`
   - `GET /api/v3/admin/client-telemetry/offline-storage/analytics?days=7` -> `200`
   - `GET /api/v3/admin/client-telemetry/offline-storage?page=0&size=5&search=admin` -> `200` với page rỗng
   - `GET /api/v3/admin/client-telemetry/offline-storage?page=0&size=5&eventType=disabled` -> `500`
4. Query DB production xác nhận bảng hiện đang thật sự chưa có bản ghi nào:

```sql
select count(*) as total, min(created_at) as first_seen, max(created_at) as last_seen
from client_offline_storage_telemetry;
```

Kết quả hiện tại:

```text
total | first_seen | last_seen
0     |            |
```

Tức là analytics `0` là đúng với dữ liệu hiện có, nhưng endpoint logs vẫn đang hỏng và không trả empty page đúng contract.

## Backend Log
Trích log production từ backend container khi mở page mặc định:

```text
Caused by: org.hibernate.exception.SQLGrammarException: JDBC exception executing SQL ...
ERROR: function lower(bytea) does not exist
Hint: No function matches the given name and argument types. You might need to add explicit type casts.
```

Stack trace đi qua query generated cho `client_offline_storage_telemetry` với điều kiện:

```sql
... lower(uje1_0.email) like lower(('%'||?||'%')) ...
```

## Code Evidence
Backend:
- `backend/src/main/java/com/example/lms/shared/infrastructure/web/ClientTelemetryControllerV3.java:63-88`
  - `listOfflineStorageTelemetry(...)` luôn gọi `telemetryJpaRepository.findFiltered(...)` với `normalize(search)`.
- `backend/src/main/java/com/example/lms/shared/infrastructure/persistence/OfflineStorageTelemetryJpaRepository.java:30-39`
  - `findFiltered(...)` dùng JPQL với điều kiện nullable search:
    - `LOWER(u.email) LIKE LOWER(CONCAT('%', :search, '%'))`
    - `LOWER(u.fullName) LIKE LOWER(CONCAT('%', :search, '%'))`

Frontend:
- `fe/src/app/features/admin/presentation/components/offline-storage-telemetry.component.ts:63-86`
  - Khi `loadLogs()` lỗi, component set `logs=[]`, `totalElements=0`, `totalPages=0`, rồi chỉ bắn toast.
- `fe/src/app/features/admin/presentation/components/offline-storage-telemetry.component.html:240-246`
  - Với `logs().length === 0`, UI render empty state `Chưa có bản ghi phù hợp`, nên không phân biệt được giữa `API 500` và `page rỗng thật`.

## Why This Matters
- Admin vận hành nhìn màn này sẽ dễ kết luận sai rằng telemetry đang sạch, trong khi endpoint logs thực tế bị hỏng.
- Khi incident offline thật xảy ra, khu vực quan trọng nhất của màn hình lại fail đúng ngay trạng thái mặc định.
- Workaround hiện tại là nhập một từ khóa tìm kiếm bất kỳ (`search=admin`) để tránh đường query lỗi, nhưng đây là hành vi ngẫu nhiên và không discoverable.

## Scope
1. Sửa backend query cho `findFiltered(...)` để không nổ khi `search = null`.
2. Giữ contract trả page rỗng (`200`, `content=[]`) khi không có bản ghi.
3. Sửa frontend để hiển thị error state rõ ràng khi `logs` request fail, thay vì rơi về empty state giống trường hợp không có dữ liệu.
4. Thêm test regression cho:
   - mặc định không `search`
   - `eventType` có giá trị nhưng `search` rỗng
   - `search` có giá trị
   - bảng telemetry rỗng hoàn toàn

## Acceptance Criteria
- Mở `/admin/offline-storage` với bộ lọc mặc định không còn gây `500`.
- `GET /api/v3/admin/client-telemetry/offline-storage?page=0&size=20` trả `200` + page rỗng khi DB không có record.
- `eventType=disabled` không còn làm endpoint nổ khi `search` rỗng.
- Frontend phân biệt được `empty` và `request failed` cho khu vực bảng logs.
- Có automated test khóa lại regression này.

## Out Of Scope
- Không cần thay đổi ingestion flow nếu chưa có bằng chứng client gửi sai.
- Không cần ép seed hoặc generate fake telemetry chỉ để lấp đầy dashboard.

--- a/backend/src/main/java/com/example/lms/shared/infrastructure/persistence/OfflineStorageTelemetryJpaRepository.java
+++ b/backend/src/main/java/com/example/lms/shared/infrastructure/persistence/OfflineStorageTelemetryJpaRepository.java
@@ -27,15 +27,36 @@ public interface OfflineStorageTelemetryJpaRepository extends JpaRepository<Offl
         Long getRecreateFailedCount();
     }
 
-    @Query("""
-        SELECT e FROM OfflineStorageTelemetryJpaEntity e
-        JOIN UserJpaEntity u ON u.id = e.userId
-        WHERE (:eventType IS NULL OR e.eventType = :eventType)
-          AND (:userId IS NULL OR e.userId = :userId)
-          AND (:search IS NULL OR LOWER(u.email) LIKE LOWER(CONCAT('%', :search, '%'))
-               OR LOWER(u.fullName) LIKE LOWER(CONCAT('%', :search, '%')))
-        ORDER BY e.createdAt DESC
-        """)
+    // Native SQL instead of JPQL: Hibernate 6 JPQL parser trips over
+    // LOWER(CONCAT('%', :search, '%')) when :search is bound to NULL, raising
+    // "operator does not exist: text like text" at runtime. Native SQL tolerates
+    // NULL in CONCAT (PostgreSQL concat treats NULL as ''), which is the same
+    // pattern every other query in this repository already uses.
+    // See issue #77 for the production 500 this caused.
+    @Query(
+        value = """
+            SELECT e.*
+            FROM client_offline_storage_telemetry e
+            JOIN users u ON u.id = e.user_id
+            WHERE (:eventType IS NULL OR e.event_type = :eventType)
+              AND (CAST(:userId AS uuid) IS NULL OR e.user_id = CAST(:userId AS uuid))
+              AND (:search IS NULL
+                   OR LOWER(u.email) LIKE LOWER(CONCAT('%', :search, '%'))
+                   OR LOWER(u.full_name) LIKE LOWER(CONCAT('%', :search, '%')))
+            ORDER BY e.created_at DESC
+            """,
+        countQuery = """
+            SELECT COUNT(*)
+            FROM client_offline_storage_telemetry e
+            JOIN users u ON u.id = e.user_id
+            WHERE (:eventType IS NULL OR e.event_type = :eventType)
+              AND (CAST(:userId AS uuid) IS NULL OR e.user_id = CAST(:userId AS uuid))
+              AND (:search IS NULL
+                   OR LOWER(u.email) LIKE LOWER(CONCAT('%', :search, '%'))
+                   OR LOWER(u.full_name) LIKE LOWER(CONCAT('%', :search, '%')))
+            """,
+        nativeQuery = true
+    )
     Page<OfflineStorageTelemetryJpaEntity> findFiltered(String eventType, UUID userId, String search, Pageable pageable);
 
     @Query(value = """

